### ISSUE TITLE
Loader.exe | CRUS_Starfighter parsing failed

### ISSUE THREAD

**Issue by Olakeen** [richardthombs/scunpacked#88]
Hi,

With the 3.15.1 build,  I have this issue with the 2 Ares files (inferno or ion) :

D:\unp4k-suite\Data\Scripts\Entities\Vehicles\Implementations\Xml\CRUS_Starfighter.xml
Unhandled exception. System.InvalidOperationException: There is an error in the XML document.
 ---> System.FormatException: Input string was not in a correct format.
   at System.Number.ThrowOverflowOrFormatException(ParsingStatus status, TypeCode type)
   at System.Double.Parse(String s, NumberStyles style, IFormatProvider provider)
   at System.Xml.XmlConvert.ToDouble(String s)
   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationReaderVehicle.Read13_Part(Boolean isNullable, Boolean checkType)
   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationReaderVehicle.Read12_DamageBehavior(Boolean isNullable, Boolean checkType)
   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationReaderVehicle.Read13_Part(Boolean isNullable, Boolean checkType)
   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationReaderVehicle.Read13_Part(Boolean isNullable, Boolean checkType)
   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationReaderVehicle.Read18_Vehicle(Boolean isNullable, Boolean checkType)
   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationReaderVehicle.Read19_Vehicle()
   --- End of inner exception stack trace ---

I didn't find what value is missing in the files.

**Comment by Olakeen** [comment]
Errors are in this file : Data\Scripts\Entities\Vehicles\Implementations\Xml\CRUS_Starfighter.xml

Still investigate

**Comment by richardthombs** [comment]
Thanks for this. I can't really work around this kind of an error unless I preprocess XML files and do a string replace :) Like what @Olakeen says, the best way to sort this out is to edit the starfighter's XML. Sadly this still isn't right in 3.16.0. I've notified CIG now, maybe they will fix it.

I have updated the vehicle parser so that we get a more helpful error so hopefully next time it will be easier to spot the broken value.
